src/main/java/bio/terra/cli/command/app/passthrough/Git.java

```diff
@@ -44,20 +54,36 @@ protected void executeImpl() {
           "Conflicted input argument. Only specify one depending on "
               + "whether you'd like to clone all or some of the git repo resources in this workspace");
     }
-    if (cloneAll) {
+    // handle terra git clone
+    if (cloneAll || (names != null && names.length > 0)) {
       validateCloneCommand();
-      Context.requireWorkspace().getResources().stream()
-          .filter(resource -> Resource.Type.GIT_REPO == resource.getResourceType())
-          .forEach(resource -> cloneGitRepoResource(resource));
-      return;
-    }
-    if (names != null) {
-      validateCloneCommand();
-      for (String name : names) {
-        cloneGitRepoResource(Context.requireWorkspace().getResource(name));
+
+      List<Resource> gitReposToClone = new ArrayList<>();
+      if (cloneAll) {
+        var resources = Context.requireWorkspace().getResources();
+        gitReposToClone.addAll(getGitReposToClone(resources));
+        resources.stream()
+            .filter(resource -> Resource.Type.DATA_COLLECTION == resource.getResourceType())
+            .forEach(
+                resource ->
+                    gitReposToClone.addAll(
+                        attemptToGetGitReposInDataCollectionToClone((DataCollection) resource)));
+      } else {
+        for (String name : names) {
```

Comment:
opt it's a bit inconsistent to use stream above, and for loop here. Can you use stream here?